pkg/odo/util/completion/completionhandlers.go

```diff
@@ -39,3 +41,57 @@ var ServiceClassCompletionHandler = func(args complete.Args, client *occlient.Cl
 
 	return completions
 }
+
+// ProjectNameCompletionHandler provides project name completion
+var ProjectNameCompletionHandler = func(args complete.Args, client *occlient.Client, command *cobra.Command) (completions []string) {
+	completions = make([]string, 0)
+	projects, err := project.List(client)
+	if err != nil {
+		return completions
+	}
+
+	var commands []string
+
+	// get only the user typed commands/flags and remove the cobra defined commands
+	found := false
+	for _, arg := range args.Completed {
+		if arg == command.Name() {
+			found = true
+			continue
+		}
+		if found {
+			commands = append(commands, arg)
+		}
+	}
+
+	// extract the flags and commands
+	strippedCommands, strippedFlags := getCommandsAndFlags(commands, command)
```

Comment:
Why not make `getCommandsAndFlags` return a map for commands directly?

Reply:
I thought of that but thought that list would be useful later. I will change it to map